src/main/java/core/basesyntax/operation/ReturnHandler.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.service.ShopService;
+import core.basesyntax.transaction.FruitTransaction;
+
+public class ReturnHandler implements OperationHandler {
+    private ShopService shopService;
+
+    public ReturnHandler(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.RETURN) {

```

Comment:
The check of operation type is redundant here. It is the responsibility of the class that uses the Strategy pattern to choose the correct strategy for each operation. You can remove this check.